main_OSEM.py

```diff
@@ -0,0 +1,102 @@
+        NB: in practice, `num_subsets` should likely be determined from the data.
+        This is just an example. Try to modify and improve it!
+        """
+        self.acquisition_models = []
+        self.prompts_subsets = []
+        self.subset_sensitivities = []
+        self.subset = 0
+        self.x = data.OSEM_image.clone()
+    
+        # find views in each subset
+        # (note that SIRF can currently only do subsets over views)
+        views = data.mult_factors.dimensions()[2]
+        partitions_idxs = partition_indices(num_subsets, list(range(views)), stagger=True)
+
+        # for each subset: find data, create acq_model, and create subset_sensitivity (backproj of 1)
+        for i in range(num_subsets):
+            prompts_subset = data.acquired_data.get_subset(partitions_idxs[i])
+            additive_term_subset = data.additive_term.get_subset(partitions_idxs[i])
+            multiplicative_factors_subset = data.mult_factors.get_subset(partitions_idxs[i])
+        
+            acquisition_model_subset = STIR.AcquisitionModelUsingParallelproj()
+            acquisition_model_subset.set_additive_term(additive_term_subset)
+            acquisition_model_subset.set_up(prompts_subset, self.x)
+
+            subset_sensitivity = acquisition_model_subset.backward(multiplicative_factors_subset)
+            # add a small number to avoid NaN in division
+            subset_sensitivity += subset_sensitivity.max() * 1e-6
+
+            self.acquisition_models.append(acquisition_model_subset)
+            self.prompts_subsets.append(prompts_subset)
+            self.subset_sensitivities.append(subset_sensitivity)
+
+        super().__init__(update_objective_interval=update_objective_interval, **kwargs)
+        self.configured = True # required by Algorithm
+
+    def update(self):
+        # compute forward projection for the denomintor
+        # add a small number to avoid NaN in division, as OSEM lead to 0/0 or worse.
+        # (Theoretically, MLEM cannot, but it might nevertheless due to numerical issues)
+        denom = self.acquisition_models[self.subset].forward(self.x) + .0001
+        # divide measured data by estimate (ignoring mult_factors!)
+        quotient = self.prompts_subsets[self.subset] / denom
+
+        # update image with quotient of the back projection (without mult_factors!) and the subset_sensitivity
+        self.x *= self.acquisition_models[self.subset].backward(quotient) / self.subset_sensitivities[self.subset]
+        self.subset = (self.subset + 1) % len(self.prompts_subsets)
+
+    def update_objective(self):
+        # should do some stuff here to have a correct TensorBoard display, but the
+        # objective function value is not part of the challenge, we will take an ugly short-cut.
+        # If you need it, you can implement it as a sum over subsets of something like
+        #   prompts * log(acq_model.forward(self.x)) - self.x * subset_sensitivity
+        return 0
```

Comment:
why not implement it?

Reply:
Because I have run out of time, while actually making the point that if they don't need it, they needn't bother implementing it either.

Feel free to implement (and test it) of course...